src/shared.rs

```diff
@@ -262,7 +262,7 @@ pub(crate) fn get_socket_path(socket_type: SocketType) -> String {
     format!("/tmp/hypr/{hypr_instance_sig}/{socket_name}")
 }
 
-pub(crate) fn object_empty_as_none<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
+pub fn object_empty_as_none<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
```

Comment:
Why are you making these public?

Reply:
Please keep this private, this is a very specific implementation detail that should not pollute the api